ChangeLog.d/fix-dtls-connection-id-compat-build.txt

```diff
@@ -0,0 +1,4 @@
+Bugfix
+   * Fix a compilation warning raised in mbedtls/ssl.h when
+     MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT is not defined which may happen
+     when MBEDTLS_SSL_DTLS_CONNECTION_ID is neither defined.
```

Comment:
This is a bit verbose and not quite correct English (“\*<s>is neither defined</s>” should be “is not defined either”). Suggested wording:

> Fix a compilation warning in ssl.h when MBEDTLS_SSL_DTLS_CONNECTION_ID and MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT are not defined.

Reply:
Ok, thanks for the suggestion.